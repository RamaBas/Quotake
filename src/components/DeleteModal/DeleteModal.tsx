import React from 'react';
import { Typography, Button, Stack } from '@mui/material';
import { useTranslation } from 'react-i18next';
import Modal from '../Modal/Modal';

interface DeleteModalProps {
  open: boolean;
  onClose: () => void;
  onConfirm: () => void;
}

const DeleteModal: React.FC<DeleteModalProps> = ({ 
  open, 
  onClose, 
  onConfirm 
}) => {
  const { t } = useTranslation();

  return (
    <Modal open={open} onClose={onClose} width={350} data-testid="delete-modal">
      <Stack spacing={3}>
        <Typography variant="h6" textAlign="center">
          {t('deleteConfirmation')}
        </Typography>
        
        <Stack direction="row" spacing={2} justifyContent="center">
          <Button 
            variant="outlined" 
            onClick={onClose}
            sx={{ flex: 1 }}
          >
            {t('denyButton')}
          </Button>
          <Button 
            variant="contained" 
            color="error"
            onClick={onConfirm}
            sx={{ flex: 1 }}
          >
            {t('confirmButton')}
          </Button>
        </Stack>
      </Stack>
    </Modal>
  );
};

export default DeleteModal;