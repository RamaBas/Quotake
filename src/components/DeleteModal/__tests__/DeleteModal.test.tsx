import { render, screen, fireEvent } from '@testing-library/react';
import DeleteModal from '../DeleteModal';

// Mock del Modal base para simplificar testing
jest.mock('../../Modal/Modal', () => ({ 
  __esModule: true,
  default: ({ open, children }: { open: boolean; children: React.ReactNode }) => 
    open ? <div data-testid="modal-mock">{children}</div> : null
}));

describe('DeleteModal Component', () => {
  const mockOnClose = jest.fn();
  const mockOnConfirm = jest.fn();
  const defaultProps = {
    open: true,
    onClose: mockOnClose,
    onConfirm: mockOnConfirm
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly when open', () => {
    render(<DeleteModal {...defaultProps} />);
    
    expect(screen.getByTestId('modal-mock')).toBeInTheDocument();

    expect(screen.getByText('deleteConfirmation')).toBeInTheDocument();
    expect(screen.getByText('denyButton')).toBeInTheDocument();
  });

  it('does not render when closed', () => {
    render(<DeleteModal {...defaultProps} open={false} />);
    
    expect(screen.queryByTestId('modal-mock')).not.toBeInTheDocument();
  });

  it('calls onClose when clicking cancel button', () => {
    render(<DeleteModal {...defaultProps} />);
    
    const cancelButton = screen.getByText('denyButton');
    fireEvent.click(cancelButton);
    
    expect(mockOnClose).toHaveBeenCalledTimes(1);
    expect(mockOnConfirm).not.toHaveBeenCalled();
  });

  it('matches snapshot when open', () => {
    const { asFragment } = render(<DeleteModal {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});