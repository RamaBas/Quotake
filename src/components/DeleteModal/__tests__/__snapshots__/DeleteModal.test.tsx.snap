// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal Component matches snapshot when open 1`] = `
<DocumentFragment>
  <div
    data-testid="modal-mock"
  >
    <div
      class="MuiStack-root css-v2woy3-MuiStack-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-19nfv5h-MuiTypography-root"
      >
        deleteConfirmation
      </h6>
      <div
        class="MuiStack-root css-14w1sp1-MuiStack-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1axjf0i-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          denyButton
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError css-nb3w9g-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          confirmButton
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
